realm-annotations-processor/src/test/java/io/realm/processor/RealmProcessorTest.java

```diff
@@ -80,6 +82,39 @@ public void compareProcessedSimpleFile() throws Exception {
     }
 
     @Test
+    public void compileSNullTypesFile() {
+        ASSERT.about(javaSource())
+                .that(nullTypesModel)
+                .compilesWithoutError();
+    }
+
+    @Test
+    public void compileProcessedNullTypesFile() throws Exception {
+        ASSERT.about(javaSource())
+                .that(nullTypesModel)
+                .processedWith(new RealmProcessor())
+                .compilesWithoutError();
+    }
+
+    @Test
+    public void compileNullTypesProxyFile() throws Exception {
```

Comment:
We don't  usually test this, the compareProcessedNullTypesFile test should be enough